src/modules/robot/Robot.cpp

```diff
@@ -599,16 +599,16 @@ void Robot::on_gcode_received(void *argument)
                 } else if(gcode->subcode == 4) {
                     // G92.4 is a smoothie special it sets manual homing for X,Y,Z
                     // do a manual homing based on given coordinates, no endstops required
-                    if(gcode->has_letter('X')){ THEROBOT->reset_axis_position(gcode->get_value('X'), X_AXIS); }
-                    if(gcode->has_letter('Y')){ THEROBOT->reset_axis_position(gcode->get_value('Y'), Y_AXIS); }
-                    if(gcode->has_letter('Z')){ THEROBOT->reset_axis_position(gcode->get_value('Z'), Z_AXIS); }
+                    if(gcode->has_letter('X')){ THEROBOT->reset_axis_position(to_millimeters(gcode->get_value('X')), X_AXIS); }
+                    if(gcode->has_letter('Y')){ THEROBOT->reset_axis_position(to_millimeters(gcode->get_value('Y')), Y_AXIS); }
+                    if(gcode->has_letter('Z')){ THEROBOT->reset_axis_position(to_millimeters(gcode->get_value('Z')), Z_AXIS); }
 
                 } else if(gcode->subcode == 3) {
                     // initialize G92 to the specified values, only used for saving it with M500
                     float x= 0, y= 0, z= 0;
-                    if(gcode->has_letter('X')) x= gcode->get_value('X');
-                    if(gcode->has_letter('Y')) y= gcode->get_value('Y');
-                    if(gcode->has_letter('Z')) z= gcode->get_value('Z');
+                    if(gcode->has_letter('X')) x= to_millimeters(gcode->get_value('X'));
```

Comment:
This is incorrect as it is only used to save to config-override, so should never be specified in inches especially as M500 will save it out in mm. Also if G20 were changed then config-override loaded all hell would break lose.